components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAOTest.java

```diff
@@ -0,0 +1,1624 @@
+            addTestIdps(idPManagementDAO);
+
+            assertThrows(IdentityProviderManagementException.class, () ->
+                    idPManagementDAO.deleteIdPByResourceId(uuid, tenantId, TENANT_DOMAIN));
+        }
+    }
+
+    @Test(dataProvider = "deleteIdPData")
+    public void testForceDeleteIdP(String idpName, int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            idPManagementDAO.forceDeleteIdP(idpName, tenantId, TENANT_DOMAIN);
+            int resultSize = getIdPCount(connection, idpName, tenantId);
+            assertEquals(resultSize, 0, "'forceDeleteIdP' method fails");
+        }
+    }
+
+    @Test(dataProvider = "deleteIdPExceptionData")
+    public void testForceDeleteIdPException(String idpName, int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            assertThrows(IdentityProviderManagementException.class, () ->
+                    idPManagementDAO.forceDeleteIdP(idpName, tenantId, TENANT_DOMAIN));
+        }
+    }
+
+    @Test(dataProvider = "deleteIdPData")
+    public void testForceDeleteIdPByResourceId(String idpName, int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            String uuid = idPManagementDAO.getIdPByName(connection, idpName, tenantId, TENANT_DOMAIN).getResourceId();
+            idPManagementDAO.forceDeleteIdPByResourceId(uuid, tenantId, TENANT_DOMAIN);
+            int resultSize = getIdPCount(connection, idpName, tenantId);
+            assertEquals(resultSize, 0, "'forceDeleteIdPByResourceId' method fails");
+        }
+    }
+
+    @Test(dataProvider = "deleteIdPExceptionData")
+    public void testForceDeleteIdPByResourceIdException(String uuid, int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
```

Comment:
Can't we have this in the setup method?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3493/commits/ef44da16e93a4a038f814603c42eadfb3a300f15